src/plugins/LADTable/components/LADRow.vue

```diff
@@ -109,56 +101,61 @@ export default {
     },
     methods: {
         updateValues(datum) {
-            let newTimestamp = datum[this.timestampKey];
+            let newTimestamp = this.formats[this.timestampKey].parse(datum),
+                update = false, limit;
 
             if(this.inBounds(newTimestamp)) {
 
-                // if current datum, else assign new current
-                if(this.currentDatum) {
-                    let existingTimestamp = this.currentDatum[this.timestampKey];
-                    // if existing is in bounds, if not assign new current
+                // if timestamp is set, need tocheck, else update
+                if(this.timestamp !== '---') {
+                    let existingTimestamp = this.formats[this.timestampKey].parse(this.timestamp);
```

Comment:
Parsing, while necessary, is expensive. Let's store the parsed value for the existing timestamp so that we only have to `.parse()` it once, when we receive it. `this.timestamp` could be the parsed value, and we could create a new computed value named `formattedTimestamp`. It would format and return `this.timestamp`. That way we're only parsing it once, and Vue can do it's optimization magic and only format it when it needs to (ie. when the underlying value changes).

Reply:
Moved this to a computed value